src/commands/utils/udfRegistration.ts

```diff
@@ -1,27 +1,104 @@
-import { Uri, window } from "vscode";
+import { ProgressLocation, Uri, window } from "vscode";
+import { udfsChanged } from "../../emitters";
+import { logError } from "../../errors";
+import { CCloudResourceLoader } from "../../loaders";
 import { Logger } from "../../logging";
+import { CCloudFlinkDbKafkaCluster } from "../../models/kafkaCluster";
+import {
+  showErrorNotificationWithButtons,
+  showInfoNotificationWithButtons,
+} from "../../notifications";
+import { flinkDatabaseQuickpick } from "../../quickpicks/kafkaClusters";
+import { logUsage, UserEvent } from "../../telemetry/events";
 import { inspectJarClasses, JarClassInfo } from "../../utils/jarInspector";
+import { FlinkDatabaseViewProvider } from "../../viewProviders/flinkDatabase";
 import { validateUdfInput } from "./uploadArtifactOrUDF";
 
 const logger = new Logger("commands.utils.udfRegistration");
+export interface UdfRegistrationData {
+  /** The class to register as a UDF */
+  classInfo: JarClassInfo;
+  /** The function name to use for the UDF */
+  functionName: string;
+}
+// internal for ease of passing it down a level
+export interface ProgressReporter {
+  report(value: { message?: string; increment?: number }): void;
+}
```

Comment:
I like the idea of specifying the interface for `{ message?: string; increment?: number }` for easier reference in the codebase, but with something like this:
```suggestion
export interface ProgressReport {
  message?: string;
  increment?: number;
}
```
We could then use the VS Code [`Progress`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bc2cd4d95b2e6a07a8da77d6b0cc2a89f8749ba2/types/vscode/index.d.ts#L7608-L7616) directly like `Progress<ProgressReport>`